client/go/vespa/target.go

```diff
@@ -139,6 +134,17 @@ type responseFunc func(status int, response []byte) (bool, error)
 
 type requestFunc func() *http.Request
 
+// waitForOK queries url and returns its status code. If the url returns a non-200 status code, it is repeatedly queried
+// until timeout elapses.
+func waitForOK(url string, certificate *tls.Certificate, timeout time.Duration) (int, error) {
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return 0, err
+	}
+	okFunc := func(status int, response []byte) (bool, error) { return status/100 == 2, nil }
```

Comment:
Can we add a `Is2xxStatus()` function and use that for the cases where we do `status/100 ==/!= 2`?
I know you just moved code around now, but I had to think a bit too hard at this pattern to understand what it was trying to do.

Reply:
It's only in one place now.

Reply:
I count it 5 places in this PR.

Reply:
I thought you meant the this particular function. PTAL last commit.